zcash_client_backend/src/data_api.rs

```diff
@@ -872,13 +872,29 @@ impl<A> ScannedBlock<A> {
 }
 
 /// A transaction that was detected during scanning of the blockchain,
-/// including its decrypted Sapling outputs.
+/// including its decrypted Sapling and/or Orchard outputs.
 ///
 /// The purpose of this struct is to permit atomic updates of the
 /// wallet database when transactions are successfully decrypted.
 pub struct DecryptedTransaction<'a, AccountId> {
-    pub tx: &'a Transaction,
-    pub sapling_outputs: &'a Vec<DecryptedOutput<sapling::Note, AccountId>>,
+    tx: &'a Transaction,
+    outputs: &'a Vec<DecryptedOutput<Note, AccountId>>,
+}
+
+impl<'a, AccountId> DecryptedTransaction<'a, AccountId> {
+    /// Constructs a new [`DecryptedTransaction`] from its constituent parts.
+    pub fn new(tx: &'a Transaction, outputs: &'a Vec<DecryptedOutput<Note, AccountId>>) -> Self {
+        Self { tx, outputs }
+    }
+
+    /// Returns the raw transaction data.
+    pub fn tx(&self) -> &Transaction {
+        self.tx
+    }
+    /// Returns the Sapling outputs that were decrypted from the transaction.
```

Comment:
```suggestion
    /// Returns the outputs that were decrypted from the transaction.
```

Reply:
Resolved in a slightly different fashion; Orchard and Sapling are kept separate at this level.